Gantt filtering[duda] (#2989)
DevExpress/devextreme-documentation: concepts/05 UI Components/Gantt/20 Filter Data/1 Filter Row.md

> The **Gantt** component allows users to filter columns by filter row values. To apply filter criteria, users enter a value in a column's filter row cell or select the value with a specific editor. 

Reviewer: ```suggestion
The **Gantt** component allows users to filter columns by filter row values. To apply filter criteria, users can enter a value in a column's filter row cell or select the value in a specific editor. 
```

Author: I agree, thank you!

> The **Gantt** component allows users to filter columns by filter row values. To apply filter criteria, users can enter a value in a column's filter row cell or select the value in a specific editor. 

Reviewer 2: `in a specific editor` Which specific editor?

Author: The filter row cells contain built-in editors (I've described this below). The type of an editor depends on a column data type. In this case, `specific` means that the control supports several editors, but you can only use one depending on the data type

Reviewer 2: To be honest, `specific` doesn't mean depending on the data type. Perhaps 'corresponding'? Even so, if there is only one editor, then you would omit it - 
```suggestion
The **Gantt** component allows users to filter columns by filter row values. To apply filter criteria, users can enter a value in a column's filter row cell or select the value in the editor. 
```

Author: ```suggestion
The **Gantt** component allows users to filter columns by filter row values. To apply filter criteria, users can enter a value in a column's filter row cell or select the value in this cell's editor.
```

> Set the **filterRow**.[visible](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/filterRow/#visible) property to **true** to display the filter row. Assign **false** to the **columns.**[allowFiltering](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/columns/#allowFiltering) property to disable the filtering feature for a column. 

Reviewer 2: ```suggestion
Set the **filterRow**.[visible](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/filterRow/#visible) property to **true** to display the filter row. Assign **false** to the **columns.**[allowFiltering](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/columns/#allowFiltering) property to disable filtering for a column. 
```

Author: Okay, thank you!

![DevExtreme HTML5 JavaScript jQuery Knockout Angular Gantt Filtering FilterRow](/images/Gantt/Visual_elements/Filter_row.png)

#include common-demobutton with {
    url: "https://js.devexpress.com/Demos/WidgetsGallery/Demo/Gantt/FilterRow/"
}

> The filter row contains a magnifier icon in its cells. Users click this icon to select the desired filter operation. Set the [showOperationChooser](/Documentation/21_2/ApiReference/UI_Components/dxGantt/Configuration/filterRow/#showOperationChooser) property to **false** for a column to hide the magnifier icon. 

Reviewer: What do you mean by "desired filter operation"? Does a click on the magnifier icon invokes a pop-up menu that allows one to choose between available options? I believe we need more clarity about this here.

```suggestion
The filter row contains a magnifier icon in its cells. Users can click this icon to select the desired filter operation. Set the [showOperationChooser](/Documentation/21_2/ApiReference/UI_Components/dxGantt/Configuration/filterRow/#showOperationChooser) property to **false** for a column to hide the magnifier icon. 
```

Author: Okay, but I'm not sure that we can say it as a pop-up menu, I believe it's better to say `list`. And `desired filter operation` is an operation that users need, in another way this is a list of comparison types

What do you think about the following:

`Users can click this icon to invoke the list of comparison types and select the necessary operation.`

Author: upd:
```suggestion
The filter row contains a magnifier icon in its cells. Users can click this icon to select a comparison type from a pop-up list. Set the [showOperationChooser](/Documentation/21_2/ApiReference/UI_Components/dxGantt/Configuration/filterRow/#showOperationChooser) property to **false** for a column to hide the magnifier icon. 
```

Reviewer: Looks good to me.

    };

   export default App;

##### ASP.NET Core Controls

    <!--Razor C#-->
    @(Html.DevExtreme().Gantt()
        .Columns(columns => {
            columns.AddFor(m => m.Title)
                .DataField("title")
                .Caption("Subject")
                .FilterValue("Scope");
            columns.AddFor(m => m.Title)
                .DataField("start")
                .Caption("Start Date")
                .AllowFiltering(false);
            @* ... *@
        })
        .FilterRow(e => {
            e.Visible(true);
        })
        // ...
    )

##### ASP.NET MVC Controls

    <!--Razor C#-->
    @(Html.DevExtreme().Gantt()
        .Columns(columns => {
            columns.AddFor(m => m.Title)
                .DataField("title")
                .Caption("Subject")
                .FilterValue("Scope");
            columns.AddFor(m => m.Title)
                .DataField("start")
                .Caption("Start Date")
                .AllowFiltering(false);
            @* ... *@
        })
        .FilterRow(e => {
            e.Visible(true);
        })
        // ...
    )

---

The **Gantt** supports a predefined set of filter operations for each data type. 

> Underlying [dataType](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/columns/#dataType) defines editor types used within filter row cells:

Reviewer: ```suggestion
Underlying [dataType](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/columns/#dataType) defines in-place editor types for filter row cells:
```

Author: Okay, I like it. Thanks!

{|
|-
! **dataType**
! **Editor**
|-

| *"string", "number", "object"*
| Text box
|-

| *"boolean"*
| Drop-down list 
|-

| *"date"*
| Date picker
|-

| *"datetime"*
| Date and time picker 
|-
|}

> You can specify necessary filter operations for each column with the [filterOperations](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/columns/#filterOperations) property.

Reviewer: Hmm... This is a somewhat strange preposition choice: "to **specify** something **with** the X..."

Did you mean something like "assign a necessary combination of filter operation to the `filterOperations` property for each column"?

Author: Yes
I use `with` as `using` here.

Reviewer: Please, do be careful with "with", for it often adds vagueness in otherwise clear sentences.

Author: Okay, what do you think about this?

`You can use the [filterOperations](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/columns/#filterOperations) property to specify necessary filter operations for each column.`

Reviewer: Looks better.

Author: ```suggestion
You can use the [filterOperations](/Documentation/ApiReference/UI_Components/dxGantt/Configuration/columns/#filterOperations) property to specify necessary filter operations for each column.
```

Reviewer: OK
